stablehlo/dialect/TypeInference.cpp

```diff
@@ -2486,12 +2486,24 @@ LogicalResult inferReduceWindowOp(
     return failure();
 
   for (size_t i = 0; i < inputArgTypes.size(); ++i) {
-    if (!inputArgTypes[i].hasRank())
+    auto inputRankedType = inputs[i].getType().dyn_cast<RankedTensorType>();
+    if (!inputRankedType) {
       inferredReturnShapes.emplace_back(inputArgTypes[i].getElementType());
-    else
-      inferredReturnShapes.emplace_back(
-          inferWindowOutputShape(inputArgTypes[i].getShape(), inferredWindow),
-          inputArgTypes[i].getElementType());
+    } else {
+      auto resultShape =
+          inferWindowOutputShape(inputArgTypes[i].getShape(), inferredWindow);
+      if (encodingToBounds(inputRankedType.getEncoding()).empty()) {
```

Comment:
Let's create a temporary variable for `encodingToBounds(inputRankedType.getEncoding())`. This is a non-trivial expression that is used multiple times.